src/WinUsb/WinUsb+WinUsbOverlapped.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Buffers;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <content>
+    /// Contains the nested <see cref="WinUsbOverlapped"/> type.
+    /// </content>
+    public static partial class WinUsb
+    {
+        /// <summary>
+        /// A managed implementation of the <see cref="Kernel32.OVERLAPPED"/> structure, used to support overlapped WinUSB I/O.
+        /// </summary>
+        private class WinUsbOverlapped : Overlapped
+        {
+            private readonly Memory<byte> buffer;
+            private readonly SafeUsbHandle handle;
+            private readonly byte pipeID;
+
+            /// <summary>
+            /// The source for completing the <see cref="Completion"/> property.
+            /// </summary>
+            private readonly TaskCompletionSource<int> completion = new TaskCompletionSource<int>();
+
+            private unsafe NativeOverlapped* native;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="WinUsbOverlapped"/> class.
+            /// </summary>
+            /// <param name="handle">
+            /// A handle to the WinUSB device on which the I/O is being performed.
+            /// </param>
+            /// <param name="pipeID">
+            /// The ID of the pipe on which the I/O is being performed.
+            /// </param>
+            /// <param name="buffer">
+            /// The buffer which is used by the I/O operation. This buffer will be pinned for the duration of
+            /// the operation.
+            /// </param>
+            public WinUsbOverlapped(SafeUsbHandle handle, byte pipeID, Memory<byte> buffer)
+            {
+                this.handle = handle ?? throw new ArgumentNullException(nameof(handle));
+                this.pipeID = pipeID;
+                this.buffer = buffer;
+            }
+
+            /// <summary>
+            /// Gets a <see cref="MemoryHandle"/> to the transfer buffer.
+            /// </summary>
+            public MemoryHandle BufferHandle { get; private set; }
```

Comment:
Avoid `public` for members that aren't really public so reviewers realize they don't have to think about public API impact of these members.